lib/Dialect/FIRRTL/Transforms/SpecializeOption.cpp

```diff
@@ -0,0 +1,65 @@
+//===- SpecializeOption.cpp -------------------------------------*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "PassDetails.h"
+#include "mlir/IR/ImplicitLocOpBuilder.h"
+#include "mlir/IR/Threading.h"
+
+using namespace mlir;
+using namespace circt;
+using namespace firrtl;
+
+namespace {
+struct SpecializeOptionPass
+    : public SpecializeOptionBase<SpecializeOptionPass> {
+  using SpecializeOptionBase::selectedCase;
+  using SpecializeOptionBase::selectedOption;
+
+  void runOnOperation() override {
+    auto circuit = getOperation();
+    if (selectedOption.empty() || selectedCase.empty()) {
+      markAllAnalysesPreserved();
+      return;
+    }
+
+    auto optionGroup = circuit.lookupSymbol<OptionOp>(selectedOption);
+    if (!optionGroup) {
+      markAllAnalysesPreserved();
+      return;
+    }
+
+    auto optionCase = optionGroup.lookupSymbol<OptionCaseOp>(selectedCase);
+    if (!optionCase) {
+      markAllAnalysesPreserved();
+      return;
+    }
+
+    mlir::parallelForEach(
+        &getContext(), circuit.getOps<FModuleOp>(), [&](auto module) {
+          module.walk([&](firrtl::InstanceChoiceOp inst) {
+            if (optionGroup.getSymNameAttr() != inst.getOptionNameAttr())
+              return;
+
+            ImplicitLocOpBuilder builder(inst.getLoc(), inst);
+            auto newInst = builder.create<InstanceOp>(
+                inst->getResultTypes(), inst.getTargetOrDefaultAttr(optionCase),
+                inst.getNameAttr(), inst.getNameKindAttr(),
+                inst.getPortDirectionsAttr(), inst.getPortNamesAttr(),
+                inst.getAnnotationsAttr(), inst.getPortAnnotationsAttr(),
+                UnitAttr{}, inst.getInnerSymAttr());
```

Comment:
Any reason why the lower-to-bind isn't also copied over?

Reply:
The instance choice op doesn't have it.